hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -934,9 +929,17 @@ public boolean isBucketEmpty(String volume, String bucket)
 
           // Entry should not be marked for delete, consider only those
           // entries.
-          if(!deletedKeySet.contains(kv.getKey())) {
-            cacheKeyMap.put(kv.getKey(), kv.getValue());
-            currentCount++;
+          String key = kv.getKey();
```

Comment:
With the change you have made here, to check the cache for each entry returned from the rockdb table iterator, do we need the `cacheKeyMap` and the earlier loop to pull all the cached entries into it? Could we simply iterate the RocksDB, check the cache for deleted values and add the valid entries to the result list?

Or, is it possible for there to be entries in the cache, which are not deleted entreis, that are not in RocksDB somehow?

Reply:
As discussed, we need to separate revisit the cache usage.

Reply:
@kerneltime Please raise a jira task to track the `cacheKeyMap` usage part, if not created yet. We need to remove `cacheKeyMap` usage as well to avoid unnecessary memory usage.

Reply:
@rakeshadr I will be filling a few Jiras for the ideas around OM performance improvement.